### PR TITLE
Add a --strip-dwarf pass

### PR DESCRIPTION
**Pull request by kripken**
This pass strips DWARF debug sections, but not other debug
sections. This is useful when emitting source maps, as we do
need the SourceMapURL section, but the DWARF sections are
not longer necessary (and we've seen a testcase where they
are massively large, so big the wasm can't even be loaded in
a browser...).

Also contains a trivial one-line fix in `--extract-function` which
was necessary to create the testcase here: that pass extracts
a function from a wasm file (like `llvm-extract`) but it didn't check
if an export already existed for the function.

### CODE REVIEW

File: src/passes/Strip.cpp [hunk 1]
```diff
@@ -62,6 +62,12 @@ Pass* createStripDebugPass() {
   });
 }
 
+Pass* createStripDWARFPass() {
+  return new Strip([&](const UserSection& curr) {
+    return curr.name.find(".debug") == 0 || curr.name.find("reloc..debug") == 0;
```
Comment (dschuff) [inline src/passes/Strip.cpp]: is there really no better way to find dwarf sections than by name?

File: test/passes/strip-dwarf.bin.txt [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+ (export "__data_end" (global $global$0))
+ (export "stackSave" (func $fimport$4))
+ (export "stackAlloc" (func $fimport$5))
+ (export "stackRestore" (func $fimport$6))
+ (export "__growWasmMemory" (func $fimport$7))
+ (func $0 (; 8 ;) (type $0) (result i32)
+  (i32.const 1024)
+ )
+ ;; custom section "sourceMappingURL", size 15
+ ;; custom section "sourceMappingURL", size 15
```
Comment (dschuff) [inline test/passes/strip-dwarf.bin.txt]: should there be 2 of these sections?

File: src/passes/Strip.cpp (hunk 1 above)
Comment (kripken) [reply src/passes/Strip.cpp]: Hmm, how else would you do it? Do they have a magic number in their content or something like that?

File: test/passes/strip-dwarf.bin.txt (hunk 2 above)
Comment (kripken) [reply test/passes/strip-dwarf.bin.txt]: Ah yes, that looks bad... seems emcc emits two of those sections atm. I'll look into it. (It's not related to this PR though.)

**Pull request merged**